src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -783,6 +783,15 @@ internal bool IsAutoProperty
                 return null;
             }
         }
+
+        private bool HasNonErrorBackingField
+        {
+            get
+            {
+                // PROTOTYPE(semi-auto-props): Implement this.
+                return IsAutoProperty;
```

Comment:
>IsAutoProperty [](http://example.com/codeflow?start=23&length=14)

I think the goal of the refactoring should be to go away from using ```IsAutoProperty``` property. Simply creating a different wrapper around it doesn't get us closer to the goal.

Reply:
@AlekseyTs Sure. I'll update the implementation, will take few days though until I can (sorry for that)

Here is what I'm thinking of:

- Introduce a three-state flag `IsNonError` in `SynthesizedBackingFieldSymbol`, for which the value meanings are:
    - `null`: The backing field was early constructed due to an initializer such that we don't know yet if it's a semi auto property or not.
    - `true`: We know this is either an auto property or a semi auto property.
    - `false`: This is an error scenario where the initializer is applied to a regular property.

- We do the binding here if this flag is null, and set it while/after binding.

Sharing my thought here to save some time in case there is a fundamental thing you don't like about the approach.

Reply:
> Here is what I'm thinking of:

Based on my comments in all use-sites, it doesn't look like we need this property.

Reply:
@AlekseyTs I replaced this property with three properties per the discussions in https://github.com/dotnet/roslyn/pull/59109#discussion_r794646947 and https://github.com/dotnet/roslyn/pull/59109#discussion_r794679566

The logic doesn't yet support semi auto properties. I'm doing that per your comment, quoted below:

> Back to the logic that is supposed to give an answer whether an initializer is allowed. For legacy auto-properties I think it could be - the set method is auto-implemented, or the set method is missing and the get method is auto-implemented. If necessary, we will adjust this logic for semi-auto properties once we start working on supporting initializers for them.